subsys/bluetooth/controller/power_clock.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <soc.h>
+#include <errno.h>
+#include <device.h>
+#include <kernel_includes.h>
+#include <clock_control.h>
+#include <ble_controller_soc.h>
+
+static int hf_clock_start(struct device *dev, clock_control_subsys_t sub_system)
+{
+	ARG_UNUSED(dev);
+
+	if (!ble_controller_hf_clock_request(NULL)) {
+		bool blocking = POINTER_TO_UINT(sub_system);
+		if (blocking)
+		{
+			bool is_running = false;
+			while (!is_running) {
+				if (ble_controller_hf_clock_is_running(&is_running)) {
+					return -EFAULT;
+				}
+			}
+		}
+		return 0;
+	} else {
+		return -EFAULT;
+	}
+}
+
+static int hf_clock_stop(struct device *dev, clock_control_subsys_t sub_system)
+{
+	ARG_UNUSED(dev);
+	ARG_UNUSED(sub_system);
+
+	if (!ble_controller_hf_clock_release()) {
```

Comment:
I would prefer `ble_controller_hf_clock_release() != 0`

Reply:
Similar for other occurrences

Reply:
Done